tests/tests_unit/test_api/test_functions.py

```diff
@@ -1147,35 +1165,53 @@ def validate_file_upload_call(*args, **kwargs):
                 ]
                 # We use splitlines to ignore line ending differences between OSs:
                 assert py_file.read().decode("utf-8").splitlines() == expected_lines
-        return FileMetadata(id=123)
+        return FileMetadata(id=123, data_set_id=data_set_id)
 
     mock = fns_api_with_mock_client._cognite_client
     mock.files.upload_bytes = validate_file_upload_call
 
-    file_id = fns_api_with_mock_client._zip_and_upload_handle(function_handle_with_reqs, name="name", external_id=xid)
+    file_id = fns_api_with_mock_client._zip_and_upload_handle(
+        function_handle_with_reqs, name="name", external_id=xid, data_set_id=data_set_id
+    )
     assert file_id == 123
 
 
-@pytest.mark.parametrize("xid, overwrite", ((None, False), ("xid", True)))
-def test__zip_and_upload_folder__call_signature(fns_api_with_mock_client, xid, overwrite):
+@pytest.mark.parametrize(
+    "xid, overwrite, data_set_id",
+    (
+        (None, False, None),
+        ("xid", True, None),
+        ("xid", True, 999),
+    ),
+)
+def test__zip_and_upload_folder__call_signature(fns_api_with_mock_client, xid, overwrite, data_set_id):
     mock = fns_api_with_mock_client._cognite_client
-    mock.files.upload_bytes.return_value = FileMetadata(id=123)
+    mock.files.upload_bytes.return_value = FileMetadata(id=123, data_set_id=data_set_id)
 
     folder = Path(__file__).parent / "function_test_resources" / "good_absolute_import"
-    file_id = fns_api_with_mock_client._zip_and_upload_folder(folder, name="name", external_id=xid)
+    file_id = fns_api_with_mock_client._zip_and_upload_folder(
+        folder, name="name", external_id=xid, data_set_id=data_set_id
+    )
     assert file_id == 123
 
     mock.files.upload_bytes.assert_called_once()
     call = mock.files.upload_bytes.call_args
     assert len(call.args) == 1 and type(call.args[0]) is bytes  # noqa: E721
-    assert call.kwargs == {"name": "name.zip", "external_id": xid, "overwrite": overwrite}
+    assert call.kwargs == {"name": "name.zip", "external_id": xid, "overwrite": overwrite, "data_set_id": data_set_id}
 
 
-@pytest.mark.parametrize("xid, overwrite", ((None, False), ("xid", True)))
-def test__zip_and_upload_folder__zip_file_content(fns_api_with_mock_client, xid, overwrite):
+@pytest.mark.parametrize(
```

Comment:
A dedicated test for this would be better. If we extend these for every new feature we add, this test is going to have very many reasons to fail, which makes it less valuable when it _does_ fail.